### PR TITLE
Feature/webbrowser

### PR DESCRIPTION
**Pull request by martinfalk**

### CODE REVIEW

**Review by petersteneteg — APPROVED** [review]

File: modules/webbrowser/src/processors/webbrowserprocessor.cpp [hunk 1]
```diff
@@ -86,27 +89,52 @@ WebBrowserProcessor::WebBrowserProcessor()
...
-                break;
-            case SourceType::WebAddress:
-                fileName_.setVisible(false);
-                url_.setVisible(true);
-                break;
+    addProperty(runJS_);
+    addProperty(js_);
+
+    runJS_.onChange([this]() {
+        browser_->GetMainFrame()->ExecuteJavaScript(js_.get(), "", 1);
```
Comment (ResearchDaniel) [inline modules/webbrowser/src/processors/webbrowserprocessor.cpp]: Script should probably be executed in the process function to make sure that the web page is finished loading when pressing the button.

**Review by ResearchDaniel — APPROVED** [review]
Nice! Might want to use a file for the javascript, but I'm fine with a simple string property as well.

File: modules/webbrowser/src/processors/webbrowserprocessor.cpp [hunk 2]
```diff
@@ -266,6 +266,11 @@ void WebBrowserProcessor::OnLoadingStateChange(CefRefPtr<CefBrowser> browser, bo
 }
 
 void WebBrowserProcessor::process() {
+    if (!isBrowserLoading_ && runJSrequested_) {
```
Comment (ResearchDaniel) [inline modules/webbrowser/src/processors/webbrowserprocessor.cpp]: No need to check isBrowserLoading, see isReady in constructor. Also, would not runJS_.isModified() work?

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]

File: modules/webbrowser/src/processors/webbrowserprocessor.cpp (hunk 2 above)
Comment (martinfalk) [reply modules/webbrowser/src/processors/webbrowserprocessor.cpp]: when relying on the isReady state, isModified should be sufficient.

Comment (martinfalk) [reply modules/webbrowser/src/processors/webbrowserprocessor.cpp]: fixed in latest commit

**Review by ResearchDaniel — APPROVED** [review]

**Pull request merged**